pkg/idp/idp_metrics.go

```diff
@@ -0,0 +1,32 @@
+package idp
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	metricNamespace = "dex_operator"
+	metricSubsystem = "idp"
+)
+
+// Gauge for secret expiry time
+var (
+	infoLabels = []string{
+		"app_name",
+		"app_namespace",
+		"app_owner",
+		"provider_type",
+		"provider_name",
+		"app_registration_name",
+	}
+
+	AppInfo = prometheus.NewGaugeVec(
```

Comment:
This is the prometheus metric. We create a row for each dex app registration. Its value is the expiry date of the client secret.